compiler/bin-js_of_ocaml/compile.ml

```diff
@@ -65,6 +74,52 @@ let output_gen ~standalone ~custom_header ~build_info ~source_map output_file f
   | `Stdout -> f stdout `Stdout
   | `Name name -> Filename.gen_file name (fun chan -> f chan `File)
 
+let find_source file =
+  match Builtins.find file with
+  | Some f -> Some (Source_map.Source_content.create (Builtins.File.content f))
+  | None ->
+      if Sys.file_exists file && not (Sys.is_directory file)
+      then
+        let content = Fs.read_file file in
+        Some (Source_map.Source_content.create content)
+      else None
+
+let build_source_map_with_offset
+    (sm : Source_map.Standard.t)
+    { Source_map.sources; names; mappings } =
+  let sources_content =
+    match sm.sources_content with
+    | None -> None
+    | Some _ -> Some (List.map ~f:find_source sources)
+  in
+  let sources =
+    List.map sources ~f:(fun filename ->
+        match Builtins.find filename with
+        | None -> filename
+        | Some _ -> Filename.concat "/builtin" filename)
+  in
+  let line_offset, mappings = Source_map.Mappings.encode_with_offset mappings in
+  ( line_offset
+  , { (sm : Source_map.Standard.t) with sources; sources_content; names; mappings } )
+
+let build_source_map_many base l =
```

Comment:
It’s not much clearer what this one does. In fact, all the code here seems to rest on what `Driver.f` does with sourcemaps, which is undocumented and takes time to figure out as it is spread across so many files. Couldn’t we document `Driver.f`?

Reply:
What make you think it depends on what `Driver.f` does ? `Driver.f` just return a `Source_map.info`.  There is no much magic there.

Reply:
My bad, I thought there was some accumulation going on in the returned source map but I misread.